app/models/category.rb

```diff
@@ -302,8 +318,24 @@ def clear_subcategory_ids
     Category.clear_subcategory_ids
   end
 
-  def top_level?
-    self.parent_category_id.nil?
+  @@has_children = DistributedCache.new("has_children")
+
+  def self.has_children?(category_id)
+    @@has_children.defer_get_set(category_id.to_s) do
+      Category.where(parent_category_id: category_id).exists?
+    end
+  end
+
+  def has_children?
+    !!id && Category.has_children?(id)
+  end
+
+  def self.clear_parent_ids
+    @@has_children.clear
+  end
+
+  def clear_parent_ids
+    Category.clear_parent_ids
```

Comment:
If we are calling `category.clear_parent_ids`, are we able to only clear the affected category instead of clearing the entire cache?

Reply:
I think we could clear only the old and new parent IDs.